### PR TITLE
feat(list): added list header files review

### PR DESCRIPTION
**Pull request by JK2Kgit**
Review functions don't merge

### CODE REVIEW

File: Queue/queue.h [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+#include <node.h>
+#include <stdbool.h>
+typedef struct QUEUE_STRUCT {
+    Node* head;
+    Node* tail;
+} Queue;
 
+Queue* queue_init(Node* node);
+
+void queue_enqueue(Queue* queue, Node* node);
```
Comment (Frendzlu) [inline Queue/queue.h]: queue_add

Comment (Frendzlu) [inline Queue/queue.h]: queue_add

File: Queue/queue.h [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+
+Node* queue_dequeue(Queue* queue);
```
Comment (Frendzlu) [inline Queue/queue.h]: queue_remove

File: Queue/queue.h [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
+
+bool queue_empty(Queue* queue);
+
+//if queue is empty free it and returns True else just returns False
+bool queue_free(Queue* queue);
```
Comment (Frendzlu) [inline Queue/queue.h]: queue_is_free

Comment (Frendzlu) [reply Queue/queue.h]: or something like that

File: Queue/queue.h [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+    Node* tail;
+} Queue;
 
+Queue* queue_init(Node* node);
+
+void queue_enqueue(Queue* queue, Node* node);
+
+Node* queue_dequeue(Queue* queue);
+
+bool queue_empty(Queue* queue);
```
Comment (Frendzlu) [inline Queue/queue.h]: queue_is_empty (the word 'is' needs to be in the function name)

File: List/node.h [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    void* data;
+    struct NODE_STRUCT* next;
+} Node;
+
+Node* node_init(void* data, Node* next);
+
+Node* node_next(Node* n);
+
+//Returns pointer to data required to be free separately due to unknown type
+void* node_free(Node* n);
```
Comment (Frendzlu) [inline List/node.h:13]: same as in other

File: List/list.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+void list_add_head(List* list, Node* node);
+
+Node* list_pop_tail(List* list);
+
+Node* list_head(List* list);
+
+Node* list_tail(List* list);
+
+bool list_empty(List* list);
```
Comment (Frendzlu) [inline List/list.h]: same

File: List/list.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+//if list is empty free it and returns True else just returns False
+bool list_free(List* list);
```
Comment (Frendzlu) [inline List/list.h]: same

File: Stack/stack.h [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+Stack* stack_init(Node* node);
+
+void stack_push(Stack* stack, Node* node);
+
+Node* stack_pop(Stack* stack);
+
+Node* stack_peek(Stack* stack);
+
+bool stack_empty(Stack* stack);
```
Comment (Frendzlu) [inline Stack/stack.h]: same

File: Stack/stack.h [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+//if stack is empty free it and returns True else just returns False
+bool stack_free(Stack* stack);
```
Comment (Frendzlu) [inline Stack/stack.h:20]: same

File: Queue/queue.h (hunk 2 above)
Comment (Frendzlu) [reply Queue/queue.h]: or queue_pop

File: Queue/queue.h (hunk 1 above)
Comment (Frendzlu) [reply Queue/queue.h]: or _push

File: Stack/stack.h [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    Node* top;
+} Stack;
+
+Stack* stack_init(Node* node);
+
+void stack_push(Stack* stack, Node* node);
+
+Node* stack_pop(Stack* stack);
+
+Node* stack_peek(Stack* stack);
```
Comment (Frendzlu) [inline Stack/stack.h]: stack_top

**Review by Frendzlu — CHANGES_REQUESTED** [review]

File: Queue/queue.h (hunk 1 above)
Comment (JK2Kgit) [reply Queue/queue.h]: ok 
queue_push

Comment (JK2Kgit) [reply Queue/queue.h]: _push

File: Queue/queue.h (hunk 2 above)
Comment (JK2Kgit) [reply Queue/queue.h]: _pop

File: Queue/queue.h (hunk 3 above)
Comment (JK2Kgit) [reply Queue/queue.h]: _empty check if empty 
_free frees memory

File: Stack/stack.h (hunk 10 above)
Comment (JK2Kgit) [reply Stack/stack.h]: ok

**Review by Frendzlu — DISMISSED** [review]
git gud

**Review by Frendzlu — APPROVED** [review]
gut

**Pull request merged**